network-api/networkapi/mozfest/migrations/0017_update_wide_images.py

```diff
@@ -0,0 +1,65 @@
+# # Generated by Django 3.0.14 on 2021-08-10 18:19
+
+from django.db import migrations
+import wagtail.core.blocks
+import wagtail.core.blocks.static_block
+import wagtail.core.fields
+import wagtail.images.blocks
+import wagtailmedia.blocks
+
+
+from networkapi.mozfest.models import MozfestPrimaryPage
+
+
+
+def update_annotated_image_blocks(qs):
+    # Loop through all pages
+    for page in qs:
+        print("\t", page)
+        # If this page doesn't have a Image that needs update, don't save it.
+        # This will speed up how fast this task can run.
+        needs_saving = False
+        # Only loop through streamfield data if there's a `body` field on the page
+        if page.body:
+            # Looping through raw_data is a List of blocks.
+            for block in page.body.raw_data:
+                # If there is a type in the blocks, and the block type is a image,
+                # check if the image has the wide setting on
+                if 'type' in block and block['type'] == "image":
+                    print(block)
+                    # If so, set the new image_width value to wide
+                    if 'wide_image' in block['value'] and block['value']['wide_image'] == True:
+                        print("Found a wide image, updating.")
+                        block['value']['image_width'] = "wide"
+                        needs_saving = True
+
+            if needs_saving:
+                # If page is published already, continue to publish it.
+                # Otherwise just save a revision for draft history.
+                if page.live:
+                    print("\t\tPage is live. Publish it")
+                    page.save_revision().publish()
+                else:
+                    print("\t\tPage is draft.")
+                    page.save_revision()
```

Comment:
```suggestion
            if needs_saving:
                page.save_revision()

                if page.live:
                    page.publish()
```

Reply:
`page` needs a revision to publish. 

```suggestion
            if needs_saving:
                revision = page.save_revision()
                
                if page.live:
                    revision.publish()
```

Reply:
Thanks for the catch everyone! I have made the update and am going to mark this as resolved.